config/application.php

```diff
@@ -55,23 +55,21 @@
 /**
  * DB settings
  */
-Config::define('DB_NAME', env('DB_NAME'));
-Config::define('DB_USER', env('DB_USER'));
-Config::define('DB_PASSWORD', env('DB_PASSWORD'));
+Config::define('DB_NAME', env('DB_NAME') ?? null);
+Config::define('DB_USER', env('DB_USER') ?? null);
+Config::define('DB_PASSWORD', env('DB_PASSWORD') ?? null);
 Config::define('DB_HOST', env('DB_HOST') ?: 'localhost');
 Config::define('DB_CHARSET', 'utf8mb4');
 Config::define('DB_COLLATE', '');
 $table_prefix = env('DB_PREFIX') ?: 'wp_';
 
 if (env('DATABASE_URL')) {
-    $database_dsn = parse_url(env('DATABASE_URL'));
-    $db_name = substr($database_dsn['path'], 1);
-    $db_host = isset($database_dsn['port']) ? $database_dsn['host'] . ":" . $database_dsn['port'] : $database_dsn['host'];
-
-    Config::define('DB_NAME', $db_name);
-    Config::define('DB_USER', $database_dsn['user']);
-    Config::define('DB_PASSWORD', $database_dsn['pass'] ?? null);
-    Config::define('DB_HOST', $db_host);
+    $dsn = (object) parse_url(env('DATABASE_URL'));
+
+    Config::define('DB_NAME', substr($dsn->path, 1));
+    Config::define('DB_USER', $dsn->user);
+    Config::define('DB_PASSWORD', $dsn->pass ?? null);
+    Config::define('DB_HOST', isset($dsn->port) ? $dsn->host . ':' . $dsn->port : $dsn->host);
```

Comment:
I find this cleaner:

```suggestion
    Config::define('DB_HOST', isset($dsn->port) ? "$dsn->host:$dsn->port" : $dsn->host);
```

Reply:
I considered that– it just isn't being used elsewhere.

Reply:
I vote for `"{$dsn->host}:{$dsn->port}"` 🤷‍♂️